crates/pathfinder/src/rpc/v02/method/get_class_at.rs

```diff
@@ -0,0 +1,263 @@
+    let definition = tx
+        .query_row(
+            "SELECT definition FROM contract_code WHERE hash=?",
+            [class],
+            |row| {
+                let data = row.get_ref_unwrap(0).as_blob()?.to_vec();
+                Ok(data)
+            },
+        )
+        .optional()
+        .context("Reading definition from database")?
+        .context("Class definition is missing")?;
+
+    Ok(definition)
+}
+
+fn get_definition_at(
+    tx: &rusqlite::Transaction<'_>,
+    block: StarknetBlocksBlockId,
+    contract: ContractAddress,
+) -> Result<Vec<u8>, GetClassAtError> {
+    let global_root = StarknetBlocksTable::get_root(tx, block)
+        .context("Reading global root from database")?
+        .ok_or(GetClassAtError::BlockNotFound)?;
+
+    let tree = GlobalStateTree::load(tx, global_root).context("Loading global state tree")?;
+    let state_hash = tree
+        .get(contract)
+        .context("Fetching contract leaf in global tree")?
+        .ok_or(GetClassAtError::ContractNotFound)?;
+
+    let definition = tx
+        .query_row(
+            "SELECT definition FROM contract_code code JOIN contract_states states ON (code.hash = states.hash) WHERE states.state_hash=?",
+            [state_hash],
+            |row| {
+                let data = row.get_ref_unwrap(0).as_blob()?.to_vec();
+                Ok(data)
+            }
+        )
+        .optional()
+        .context("Reading definition from database")?
+        .context("Class definition is missing")?;
+
+    Ok(definition)
+}
+
+/// Returns the [ClassHash] of the given [ContractAddress] if any is defined in the pending data.
+async fn get_pending_class_hash(
+    pending: Option<crate::state::PendingData>,
+    address: ContractAddress,
+) -> Option<ClassHash> {
+    pending?.state_update().await.and_then(|state_update| {
+        state_update
+            .state_diff
+            .deployed_contracts
+            .iter()
+            .find_map(|contract| (contract.address == address).then_some(contract.class_hash))
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{starkhash, starkhash_bytes};
+    use assert_matches::assert_matches;
+
+    mod parsing {
+        use crate::core::StarknetBlockHash;
+
+        use super::*;
+
+        use jsonrpsee::types::Params;
+
+        #[test]
+        fn positional_args() {
+            let positional = r#"[
+                { "block_hash": "0xabcde" },
+                "0x12345"
+            ]"#;
+            let positional = Params::new(Some(positional));
+
+            let input = positional.parse::<GetClassAtInput>().unwrap();
+            let expected = GetClassAtInput {
+                block_id: StarknetBlockHash(starkhash!("0abcde")).into(),
+                contract_address: ContractAddress::new_or_panic(starkhash!("012345")),
+            };
+            assert_eq!(input, expected);
+        }
+
+        #[test]
+        fn named_args() {
+            let positional = r#"{
```

Comment:
This seems to be a common pattern in a lot of method tests, but I think the name `positional` is misleading here.

Reply:
Thanks - I really should have just named them `args` :D